cmake/target/target.cmake

```diff
@@ -135,60 +179,34 @@ endfunction(setup_single_target)
 # - SOURCES: sources specified with `set(SOURCE_FILES ...)` in module's CMakeLists.txt
 # - DEPENDENCIES: dependencies and link libraries specified with `set(MOD_DEPS ...)` in module's CMakeLists.txt
 ####
-function(setup_module_targets MODULE SOURCES DEPENDENCIES)
+function(setup_module_targets BUILD_TARGET)
```

Comment:
`target` is super overloaded here - since this is an internal function, thoughts on renaming some of those things?

BUILD_TARGET is actually a module/library name, and `setup_module_targets` actually sets up module with "F´ targets" such as `build`. Maybe we could speak of "target factories/generators(/else?)" instead ?

Reply:
Maybe for a follow up ticket

Reply:
Follow-up ticket.